Code/Framework/AzCore/AzCore/Settings/CommandLine.cpp

```diff
@@ -43,6 +47,19 @@ namespace AZ
         }
     }
 
+    struct CommandLine::ArgumentParserState
+    {
+        //! Set to true if the psuedo argument '--' has been parsed
+        //! This indicates to the argument parser that the remaining
+        //! arguments should be treated as positional arguments only
+        bool m_parseRemainAsPositional{ false};
+        //! Stores the current option being parsed
+        //! This is used for maintaining state for space separated options
+        //! Ex. --foo bar
+        //! When the "bar" argument is parsed the current switch would be set to "foo"
+        AZStd::string m_currentSwitch;
```

Comment:
question: Thinking can we change it to `m_currentOption`. Is 'switch' still needed to indicate something more than 'option' does?

Reply:
I'll change it to the option.